lib/jsdom/living/structured-clone/structuredDeserializeWithTransfer.js

```diff
@@ -0,0 +1,448 @@
+     * targetRealm whose [[ArrayBufferData]] internal slot value is serialized.[[ArrayBufferData]], whose
+     * [[ArrayBufferByteLength]] internal slot value is serialized.[[ArrayBufferByteLength]], and whose
+     * [[ArrayBufferMaxByteLength]] internal slot value is a serialized.[[ArrayBufferMaxByteLength]].
+     *
+     * If this throws an exception, catch it, and then throw a "DataCloneError" DOMException.
+     * This step might throw an exception if there is not enough memory available to create such an ArrayBuffer object.
+     */
+
+    try {
+      value = new ArrayBuffer(serialized.ArrayBufferByteLength, { maxByteLength: serialized.ArrayBufferMaxByteLength });
+      value.set(serialized.ArrayBufferData, 0);
+    } catch (err) {
+    // eslint-disable-next-line no-undef
+      throw DOMException.create(targetRealm, [
+        "The object can not be cloned.",
+        "DataCloneError"
+      ]);
+    }
+  } else if (serialized.Type === "ArrayBufferView") {
+    // Otherwise, if serialized.[[Type]] is "ArrayBufferView", then:
+    // Let deserializedArrayBuffer be
+    // ? StructuredDeserialize(serialized.[[ArrayBufferSerialized]], targetRealm, memory).
+
+    const deserializedArrayBuffer = structuredDeserialize(serialized.ArrayBufferSerialized, targetRealm, memory);
+
+    if (serialized.Constructor === "DataView") {
+      // If serialized.[[Constructor]] is "DataView", then set value to a new DataView object in targetRealm whose
+      // [[ViewedArrayBuffer]] internal slot value is deserializedArrayBuffer, whose [[ByteLength]] internal slot
+      // value is serialized.[[ByteLength]], and whose [[ByteOffset]] internal slot value is serialized.[[ByteOffset]].
+
+      value = new DataView(deserializedArrayBuffer, serialized.ByteOffset, serialized.ByteLength);
+    } else {
+      // Otherwise, set value to a new typed array object in targetRealm, using the constructor given by
+      // serialized.[[Constructor]], whose [[ViewedArrayBuffer]] internal slot value is deserializedArrayBuffer,
+      // whose [[TypedArrayName]] internal slot value is serialized.[[Constructor]], whose [[ByteLength]] internal
+      // slot value is serialized.[[ByteLength]], whose [[ByteOffset]] internal slot value is serialized.[[ByteOffset]],
+      // and whose [[ArrayLength]] internal slot value is serialized.[[ArrayLength]].
+
+      // eslint-disable-next-line no-undef
+      value = Reflect.construct(targetRealm[serialized.Constructor], [
+        deserializedArrayBuffer,
+        serialized.ByteOffset,
+        serialized.ByteLength
+      ]);
+    }
+  } else if (serialized.Type === "Map") {
+    /**
+     * Otherwise, if serialized.[[Type]] is "Map", then:
+     *
+     * Set value to a new Map object in targetRealm whose [[MapData]] internal slot value is a new empty List.
+     * Set deep to true.
+     */
+    value = new Map();
```

Comment:
Here and elsewhere, you're going to need to use `targetRealm.Map`, etc. That will fix some of the test failures.

Reply:
Is this valid for all the constructors instructions? I mean, included `ArrayBuffer` and `UInt8Array`?

Reply:
May it be better if I use `Reflect.construct`?